extensions/wikia/Blogs/BlogArticle.php

```diff
@@ -78,24 +83,21 @@ public function view() {
 	 * @access private
 	 */
 	private function showBlogListing() {
-		global $wgOut, $wgRequest, $wgMemc, $wgParser;
+		$wg = F::app()->wg;
+		$wg->Out->setSyndicated( true );
 
-		/**
-		 * use cache or skip cache when action=purge
-		 */
-		$user    = $this->mTitle->getBaseText();
-		$userMem = $this->mTitle->getPrefixedDBkey();
```

Comment:
So mTitle here will be a user's blog page (User_blog:Garthwebb), or a post (User_blog:Garthwebb/my_fantastic_post) and getBaseText returns the base title, e.g. no namespace and no subpage, i.e., 'Garthwebb'. I created getBlogOwner that encapsulates the call to getBaseText to make this a bit more clear.

The getPrefixedDBkey was misguided update in 2009 to be unnecessarily specific by adding the namespace portion User_blog. However this method returns the subpage which means that when the clearBlogListing method below tried to clear this cache, it was trying to clear the "User_blog:Garthwebb/my_fantastic_post" URL which is not the "User_blog:Garthwebb" blog listing URL that the listing was originally cached under.

This simplifies things and fixes the caching issues.